### PR TITLE
Main

### PR DESCRIPTION
**Pull request by usaotearoa**
Added 3 examples:
subscriber
publisher and subscriber
service: addtwoints

### CODE REVIEW

**Comment by usaotearoa** [comment]
@Acuadros95 
Can this PR be approved?

**Comment by usaotearoa** [comment]
This is awaiting approval,
Simple button click!

File: examples/micro-ros_addtwoints_service/platformio.ini [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+;   Upload options: custom upload port, speed and extra flags
+;   Library options: dependencies, extra library storages
+;   Advanced options: extra scripting
+;
+; Please visit documentation for the other options and examples
+; https://docs.platformio.org/page/projectconf.html
+
+[env:teensy41]
+platform = teensy
+board = teensy41
```
Comment (Acuadros95) [inline examples/micro-ros_addtwoints_service/platformio.ini:13]:
```suggestion
; [env:teensy41]
; platform = teensy
; board = teensy41
```

Examples should be board agnostic, leaving this lines commented so users can easily modify them.

Same for the rest of `platformio.ini`

File: examples/micro-ros_addtwoints_service/src/Main.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  // create executor
+  RCCHECK(rclc_executor_init(&executor, &support.context, 1, &allocator));
+  RCCHECK(rclc_executor_add_service(&executor, &service, &req, &res, service_callback));
+}
+
+
+void loop() {
+  delay(100);
+  RCSOFTCHECK(rclc_executor_spin_some(&executor, RCL_MS_TO_NS(100)));}
```
Comment (Acuadros95) [inline examples/micro-ros_addtwoints_service/src/Main.cpp:63]:
```suggestion
  RCSOFTCHECK(rclc_executor_spin_some(&executor, RCL_MS_TO_NS(100)));
}
```

style

File: examples/micro-ros_subscriber/src/Main.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  RCCHECK(rclc_subscription_init_default(
+    &subscriber,
+    &node,
+    ROSIDL_GET_MSG_TYPE_SUPPORT(std_msgs, msg, Int32),
+    "micro_ros_arduino_subscriber"));
+
+  // create executor
+  RCCHECK(rclc_executor_init(&executor, &support.context, 1, &allocator));
+  // RCCHECK(rclc_executor_add_timer(&executor, &timer));
+  RCCHECK(rclc_executor_add_subscription(&executor, &subscriber, &msg, &subscription_callback, ON_NEW_DATA));
```
Comment (Acuadros95) [inline examples/micro-ros_subscriber/src/Main.cpp:67]:
```suggestion
  RCCHECK(rclc_executor_add_subscription(
    &executor, 
    &subscriber, 
    &msg, 
    &subscription_callback, 
    ON_NEW_DATA));
```

File: examples/micro-ros_subscriber/src/Main.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  // create subscriber
+  RCCHECK(rclc_subscription_init_default(
+    &subscriber,
+    &node,
+    ROSIDL_GET_MSG_TYPE_SUPPORT(std_msgs, msg, Int32),
+    "micro_ros_arduino_subscriber"));
+
+  // create executor
+  RCCHECK(rclc_executor_init(&executor, &support.context, 1, &allocator));
+  // RCCHECK(rclc_executor_add_timer(&executor, &timer));
```
Comment (Acuadros95) [inline examples/micro-ros_subscriber/src/Main.cpp:66]:
```suggestion
```

File: examples/micro-ros_addtwoints_service/src/Main.cpp [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+#include <Wire.h>
+#include <micro_ros_platformio.h>
+#include <example_interfaces/srv/add_two_ints.h>
+#include <stdio.h>
+#include <rcl/error_handling.h>
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
+#include <std_msgs/msg/int64.h>
+
+
```
Comment (Acuadros95) [inline examples/micro-ros_addtwoints_service/src/Main.cpp:15]:
```suggestion
#if !defined(MICRO_ROS_TRANSPORT_ARDUINO_SERIAL)
#error This example is only avaliable for Arduino framework with serial transport.
#endif

```

File: examples/micro-ros_addtwoints_service/src/Main.cpp [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  allocator = rcl_get_default_allocator();
+
+  // create init_options
+  RCCHECK(rclc_support_init(&support, 0, NULL, &allocator));
+
+  // create node
+  RCCHECK(rclc_node_init_default(&node, "add_twoints_client_rclc", "", &support));
+
+  // create service
+  RCCHECK(rclc_service_init_default(&service, &node, ROSIDL_GET_SRV_TYPE_SUPPORT(example_interfaces, srv, AddTwoInts), "/addtwoints"));
```
Comment (Acuadros95) [inline examples/micro-ros_addtwoints_service/src/Main.cpp:53]:
```suggestion
  RCCHECK(rclc_service_init_default(
    &service, 
    &node, 
    ROSIDL_GET_SRV_TYPE_SUPPORT(example_interfaces, srv, AddTwoInts), 
    "micro_ros_platformio_node_service"));
```

File: examples/micro-ros_addtwoints_service/src/Main.cpp [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  set_microros_serial_transports(Serial);
+  delay(1000); 
+
+  allocator = rcl_get_default_allocator();
+
+  // create init_options
+  RCCHECK(rclc_support_init(&support, 0, NULL, &allocator));
+
+  // create node
+  RCCHECK(rclc_node_init_default(&node, "add_twoints_client_rclc", "", &support));
```
Comment (Acuadros95) [inline examples/micro-ros_addtwoints_service/src/Main.cpp:50]:
```suggestion
  RCCHECK(rclc_node_init_default(&node, "micro_ros_platformio_node", "", &support));
```

File: examples/micro-ros_addtwoints_service/src/Main.cpp [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+  // create service
+  RCCHECK(rclc_service_init_default(&service, &node, ROSIDL_GET_SRV_TYPE_SUPPORT(example_interfaces, srv, AddTwoInts), "/addtwoints"));
+
+  // create executor
+  RCCHECK(rclc_executor_init(&executor, &support.context, 1, &allocator));
+  RCCHECK(rclc_executor_add_service(&executor, &service, &req, &res, service_callback));
```
Comment (Acuadros95) [inline examples/micro-ros_addtwoints_service/src/Main.cpp:57]:
```suggestion
  RCCHECK(rclc_executor_add_service(
    &executor, 
    &service, 
    &req, 
    &res, 
    service_callback));
```

File: examples/micro-ros_pub_sub/src/Main.cpp [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+  delay(2000);
+
+  allocator = rcl_get_default_allocator();
+
+  //create init_options
+  RCCHECK(rclc_support_init(&support, 0, NULL, &allocator));
+
+  // create node
+  RCCHECK(rclc_node_init_default(&node, "uros_platformio_node", "", &support));
```
Comment (Acuadros95) [inline examples/micro-ros_pub_sub/src/Main.cpp:72]:
```suggestion
  RCCHECK(rclc_node_init_default(&node, "micro_ros_platformio_node", "", &support));
```

**Comment by Acuadros95** [comment]
Closing as some of the review comments were not resolved.

**Pull request closed without merging**